ibis/backends/tests/test_aggregation.py

```diff
@@ -1347,11 +1346,14 @@ def test_topk_op(alltypes, df):
 @pytest.mark.notyet(
     ["flink"], raises=Py4JError, reason="Flink doesn't support semi joins"
 )
-def test_topk_filter_op(alltypes, df, result_fn, expected_fn):
+def test_topk_filter_op(con, alltypes, df, result_fn, expected_fn):
     # TopK expression will order rows by "count" but each backend
     # can have different result for that.
     # Note: Maybe would be good if TopK could order by "count"
     # and the field used by TopK
+    if con.name == "sqlite":
+        # TODO: remove after CTE extraction is reimplemented
+        pytest.skip("topk -> semi-join performance has increased post SQLGlot refactor")
```

Comment:
Is this still an issue?

Reply:
Still slow, so yes it's still an issue.

Reply:
As you noted below, yes, this still is slow even with CTE extraction. I'm not sure what the it is about how we're generating this query now that doesn't play well with sqlites optimizer.